source/docs/appendices/migration-guides.md

```diff
@@ -9,7 +9,47 @@ Our goal is to make migrations as smooth as possible. This is why we try to make
 
-Nothing yet!
+### Node Version Support Change
+
+We now support node versions 14, 16, 17. So if you are using node 12 you will need to update your node version to one supported (16 recommended).
+
+In order to support node 17 we made changes that might impact your codebase:
+
+- `@apollo/*` use is prohibited. `@apollo/client` is only installed in Front-Commerce because it is needed by one of our dependencies. Other `@apollo/*` dependencies are not a `front-commerce` direct dependency and therefore should not be used. To use `2.15.0` you need to remove all uses of `@apollo/*` usages in your codebase and replace them with `react-apollo`, or `graphql-tag` imports where applicable. Check [here](https://gitlab.com/front-commerce/front-commerce/-/merge_requests/1215/diffs?commit_id=8fae8438d46200cbd310b16c30b3078da28b7d42) and [here](https://gitlab.com/front-commerce/front-commerce/-/merge_requests/1215/diffs?commit_id=418a291f917660b18672b6be2eb2778cb51dfef9)/ e.g:
+
+```diff
+-import { useLazyQuery, useQuery, useMutation } from "@apollo/react-hooks";
++import { useLazyQuery, useQuery, useMutation } from "react-apollo";
+```
+
+- resolvers for fields not in the `schema` now throws errors. To avoid this remove all unneeded `field`/`Type` resolvers that do not have a representation in the GraphQL `schema`. Check our [cleaning commit](https://gitlab.com/front-commerce/front-commerce/-/merge_requests/1216/diffs?commit_id=1e2e04e7f28dba2ebd36da50701478a13a58a238) regarding this issue.
+- `makeApolloClientStub` now expects resolvers as an argument instead of mocks. This in in tandem with the [`graphql-tools` upgrade notes](https://www.graphql-tools.com/docs/mocking#migration-from-v7-and-below). Note that we provided an adapter that converts `mocks` to `resolvers` so that old tests are not affected. However it is recommended to make the change. To disable the adapter's warning set `FRONT_COMMERCE_WEB_PRINT_MAKEAPOLLOCLIENTSTUB_MOCKS_DEPRECATION_WARNING` environment variable to `"false"`.
+- The use of `SchemaVisitor`/`SchemaDirectiveVisitor` to implement GraphQL directive is deprecated and it is replaced by the use of the [new API](https://www.graphql-tools.com/docs/schema-directives). We implemented an adapter so that uses of `SchemaVisitor`/`SchemaDirectiveVisitor` will not break, however it is recommended to make the change. Please note that the adapter only supports the use of `visitFieldDefinition` function of the `SchemaVisitor`/`SchemaDirectiveVisitor` class. If you have use for other functions and cannot make the change to the new API please contact us for help.
+- When using node 17 you need to prefix your npm commands by `NODE_OPTIONS=--openssl-legacy-provider` e.g. `NODE_OPTIONS=--oenssl-legacy-provider npm start`, `NODE_OPTIONS=--openssl-legacy-provider npm run build`.
+- When using npm >= 8.5 you need to add `--legacy-peer-deps` to npm install commands. e.g. `npm install --legacy-peer-deps`, `npm i some_dependency --legacy-peer-deps`.
```

Comment:
Are this line and the previous one  also documented in the readme and on the [installation documentation page](https://developers.front-commerce.com/docs/essentials/installation.html#Installation-using-the-skeleton) ?

Reply:
good point

Reply:
done please check